library/src/main/java/com/owncloud/android/lib/resources/notifications/GetNotificationRemoteOperation.kt

```diff
@@ -0,0 +1,105 @@
+ *   MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ *   NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ *   BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ *   ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ *   CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *   THE SOFTWARE.
+ *
+ */
+package com.owncloud.android.lib.resources.notifications
+
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import com.google.gson.JsonParser
+import com.google.gson.reflect.TypeToken
+import com.nextcloud.common.NextcloudClient
+import com.nextcloud.operations.GetMethod
+import com.owncloud.android.lib.common.operations.RemoteOperation
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.common.utils.Log_OC
+import com.owncloud.android.lib.resources.notifications.models.Notification
+import org.apache.commons.httpclient.HttpStatus
+import org.json.JSONException
+import java.io.IOException
+
+/**
+ * Provides the remote notifications from the server handling the following data structure accessible via the
+ * notifications endpoint at {@value OCS_ROUTE_LIST_V12_AND_UP}, specified at {@link
+ * "https://github.com/nextcloud/notifications/blob/master/docs/ocs-endpoint-v2.md"}.
+ */
+class GetNotificationRemoteOperation(private val id: Int) : RemoteOperation<Notification>() {
+    override fun run(client: NextcloudClient): RemoteOperationResult<Notification> {
+        var result: RemoteOperationResult<Notification>
+        val status: Int
+        var get: GetMethod? = null
+        val url = client.baseUri.toString() + OCS_ROUTE_LIST_V12_AND_UP + id + JSON_FORMAT
+
+        // get the notifications
+        try {
+            get = GetMethod(url, true)
+            status = client.execute(get)
+            val response = get.getResponseBodyAsString()
+            if (isSuccess(status)) {
+                result = RemoteOperationResult(true, get)
+                Log_OC.d(TAG, "Successful response: $response")
+
+                // Parse the response
+                result.setResultData(parseResult(response))
+            } else {
+                result = RemoteOperationResult(false, get)
+                Log_OC.e(TAG, "Failed response while getting user notifications ")
+                Log_OC.e(TAG, "*** status code: $status ; response message: $response")
+            }
+        } catch (e: IOException) {
+            result = RemoteOperationResult(e)
+            Log_OC.e(TAG, "Exception while getting remote notifications", e)
+        } finally {
+            get?.releaseConnection()
+        }
+        return result
+    }
+
+    @Throws(JSONException::class)
+    private fun parseResult(response: String): Notification {
+        val jo = JsonParser.parseString(response) as JsonObject
+        val jsonDataObject = jo.getAsJsonObject(NODE_OCS).getAsJsonObject(NODE_DATA)
+        val gson = Gson()
```

Comment:
Just a comment for future reference: we should be trying to reuse `Gson` instances throughout different operations, etc (via injection), as Gson instantiation is kinda heavy performance-wise.